tokio/src/io/util/async_write_ext.rs

```diff
@@ -1231,6 +1231,17 @@ cfg_io_util! {
         /// It is considered an error if not all bytes could be written due to
         /// I/O errors or EOF being reached.
         ///
+        /// # Cancel safety
+        ///
+        /// If `flush` is used as the event in a [`tokio::select!`](crate::select)
+        /// statement and some other branch completes first, then the data in the
+        /// buffered data in this `AsyncWrite` may have been partially flushed.
+        /// However, it is guaranteed that the buffer is advanced by the amount
+        /// of bytes that have been partially flushed.
+        ///
+        /// The same can be said in the case where the `flush` results in an error
+        /// after polling the future to completion.
```

Comment:
I don't think we can say anything about errors. If an error happens, the stream is probably in an invalid state. Probably just remove this.